src/Propulsion.SqlStreamStore/Locked.fs

```diff
@@ -0,0 +1,71 @@
+namespace Propulsion.SqlStreamStore
+
+open System
+open SqlStreamStore
+open Medallion.Threading.Sql
+
+module Locked =
+
+    /// Creates a distributed lock that manages its own db connection.
+    let distributedLock (connString: string) (consumerGroup: string) =
+        let lockName = SqlDistributedLock.GetSafeLockName(sprintf "lock-%s" consumerGroup)
+        SqlDistributedLock(lockName, connString, SqlDistributedLockConnectionStrategy.Azure)
```

Comment:
should the Azure go up into one of the names to make it clear that this is tuned for / only considers Azure?

Reply:
I've actually pulled out the locking specific parts from the PR. It mostly addresses a redundancy problem that is not core to what Propulsion wants to achieve, so I don't feel it's worth having it here. Check the update.

Reply:
👍 makes sense